src/network/protocol/messages/block.zig

```diff
@@ -0,0 +1,201 @@
+const std = @import("std");
+const native_endian = @import("builtin").target.cpu.arch.endian();
+const protocol = @import("../lib.zig");
+
+const ServiceFlags = protocol.ServiceFlags;
+
+const read_bytes_exact = @import("../../../util/mem/read.zig").read_bytes_exact;
+
+const Endian = std.builtin.Endian;
+const Sha256 = std.crypto.hash.sha2.Sha256;
+
+const CompactSizeUint = @import("bitcoin-primitives").types.CompatSizeUint;
+const BlockHeader = @import("../../../types/block.zig").BlockHeader;
+const Transaction = @import("../../../types/transaction.zig").Transaction;
+
+/// BlockMessage represents the "block" message
+///
+/// https://developer.bitcoin.org/reference/p2p_networking.html#block
+pub const BlockMessage = struct {
+    block_header: BlockHeader,
+    txn_count: CompactSizeUint,
```

Comment:
you can get rid of this fields as it will contain the same information as `ArrayList.items.len`